src/toolkits/object_detection/object_detector.cpp

```diff
@@ -778,17 +796,27 @@ void object_detector::perform_predict(gl_sframe data,
     // below should be concurrent with the neural net inference for that batch.
     pop_until_size(1);
 
-    image_augmenter::result result_batch;
+    image_augmenter::result_with_dimension result_with_dimension_batch;
 
     // Instead of giving the ground truth data to the image augmenter and the
     // neural net, instead save them for later, pairing them with the future
     // predictions.
-    result_batch.annotations_batch.resize(input_batch.size());
+    result_with_dimension_batch.result_batch.annotations_batch.resize(
+        input_batch.size());
     for (size_t i = 0; i < input_batch.size(); ++i) {
-      result_batch.annotations_batch[i] = std::move(input_batch[i].annotations);
+      result_with_dimension_batch.result_batch.annotations_batch[i] =
+          std::move(input_batch[i].annotations);
       input_batch[i].annotations.clear();
     }
 
+    // Store image dimentions
+    result_with_dimension_batch.image_dimensions_batch.resize(
+        input_batch.size());
+    for (size_t i = 0; i < input_batch.size(); ++i) {
+      result_with_dimension_batch.image_dimensions_batch[i] = std::make_pair(
```

Comment:
Nit: in most NN-specific code, height comes before width. Let’s preserve that ordering to minimize confusion, especially since we’re just using a tulle here, which doesn’t name its elements

Reply:
good!